### DISCUSSION TITLE
What checks that a stripe subscription has lapsed?

### DISCUSSION THREAD

**Discussion by woahdae** [pay-rails/pay#923] [General]
I'm writing system tests for a Pay implementation, and I'm trying to write a test covering if someone's Stripe subscription has gone past due. I assume a charge fails and they send a webhook event, but I don't see any tests in Pay except in Braintree covering this case. Is this how it works in Stripe? We rely on Stripe to tell us to disable an account? I was expecting to see `ends_at` set to match `current_period_end`, but this makes enough sense, I just would like to confirm since I can't find it in the Stripe code.

**Comment by excid3** [comment]
You can use Stripe's test clocks to forward time and create a past due subscription and use that JSON for the object with the `Pay::Stripe::Subscription.sync` method like we do in our test suite.

**Reply by woahdae** [reply]
Cool, thanks. In real life, Stripe will send a past due notice? Or is there a task that I need to run to sync subscriptions?

**Reply by excid3** [reply]
subscription.updated webhook
